www/src/public/js/controllers/builder/main.js

```diff
@@ -146,27 +147,34 @@
             // load lists
             if ($cookies.get("cookie-consent")) {
                 var listVersion = -1;
-                var listCookieStr = localStorage.getItem("cl2");
-                if (listCookieStr) {
-                    listVersion = 2;
-                }
-                else {
-                    var listCookieStr = localStorage.getItem("cl");
-                    if (!listCookieStr) {
-                        listCookieStr = $cookies.get("cl1");
-                    }
+                var listCookieStr = localStorage.getItem("cl3")
```

Comment:
There's a lot of duplicate work happening here that's quite inefficient.

On this line you're getting cl3, and then not doing anything with it, you should immediately check `if (listCookieStr)` and set it to version 3 if it's successful. In each of the other if statements you're also doing double the work. For example, on line 153, you're getting cl2 to see if it exists, and then you're setting the listCookieStr on the next line, by getting it from storage again. `localStorage.getItem()` is not a free call, so when you get it the first time, make sure you save it to the listCookieStr to reuse it later.

Additionally, you have all of these in if statements, but not else statements. Meaning that even if there is a cl3 item, it will also load up the cl1 and cl2 items (twice). All in all, there are always 7 calls to `getItem` here, when there should be at minimum 1 (they have a cl3) and at maximum 4 (they still have a cl1 for some reason). You're also missing a check for `cl`, the list without a version. While highly unlikely, it's still possible that someone has this.

Reply:
I believe I have this resolved.  Tested and working with v2 and v3's and hopefully fulfilling your desires.  I'm not sure on the original version stuff...I may have some bad checks there so let me know.